### PR TITLE
Update exit entity label logic

### PR DESCRIPTION
**Pull request by 84634E1A607A**
Found a bug when testing engulfment revamp.

**NOTICE: Because the CallbackConverter.cs fails to handle static class callback, save will fail when you turn entity label on.**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by Kasterisk** [comment]
So just to confirm, this save error only happens when you turn the entity labels on but none otherwise?

> NOTICE: Because the CallbackConverter.cs fails to handle static class callback, save will fail when you turn entity label on.

Well that's pretty interesting, I think we should either directly fix the source that is to make CallbackConverter support callbacks in static classes or until that's done omit the use of callbacks in `DebugOverlays.EntityLabel` for the time being.

**Comment by 84634E1A607A** [comment]
Previously I forgot to remove the delegate when entity label was turned off. This PR fix that, So once you close the entity label the save will be OK.

**Comment by 84634E1A607A** [comment]
Well having thought for a while, I think debug overlays should never interfere the game content. I will make another pull request, maybe it tomorrow to remove that code. Checking the status of entities every frame would not be too costy.

**Comment by hhyyrylainen** [comment]
I'll try to find the time to make a PR soon to make CallbackConverter fail on save when trying to save a delegate that points to a static method.

**Comment by 84634E1A607A** [comment]
Should be okay.

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 1]
```diff
@@ -22,6 +23,49 @@ private bool ShowEntityLabels
         }
     }
 
+    private void UpdateLabelColour(IEntity iEntity, Spatial entity, Label label)
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:26]: Why not access the Spatial node from IEntity's `EntityNode`? That way there don't have to be these 2 similar params.

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 2]
```diff
@@ -22,6 +23,49 @@ private bool ShowEntityLabels
...
+    {
+        if (!iEntity.AliveMarker.Alive)
+        {
+            label.Set("custom_colors/font_color", new Color(1.0f, 0.3f, 0.3f));
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:30]: Prefer using the explicit `AddColorOverride` as it's most often used in the codebase.

**Review by Kasterisk — CHANGES_REQUESTED** [review]
Can confirm no more save failure when testing this branch on top of the engulf revamp branch. Some suggestions:

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 3]
```diff
@@ -22,6 +23,52 @@ private bool ShowEntityLabels
         }
     }
 
+    private void UpdateLabelColour(IEntity iEntity, Label label)
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:26]:
```suggestion
    private void UpdateLabelColour(IEntity entity, Label label)
```

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 4]
```diff
@@ -22,6 +23,52 @@ private bool ShowEntityLabels
...
+    {
+        var entity = iEntity.EntityNode;
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:28]:
```suggestion
        var node = iEntity.EntityNode;
```

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 5]
```diff
@@ -32,11 +79,14 @@ private void UpdateEntityLabels()
 
         foreach (var pair in entityLabels)
         {
-            var entity = pair.Key.EntityNode;
+            var iEntity = pair.Key;
+            var entity = iEntity.EntityNode;
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:83]: No longer needed.

File: src/engine/DebugOverlays.EntityLabel.cs [hunk 6]
```diff
@@ -32,11 +79,14 @@ private void UpdateEntityLabels()
 
         foreach (var pair in entityLabels)
         {
-            var entity = pair.Key.EntityNode;
+            var iEntity = pair.Key;
```
Comment (Kasterisk) [inline src/engine/DebugOverlays.EntityLabel.cs:82]:
```suggestion
            var entity = pair.Key;
```

**Review by Kasterisk — CHANGES_REQUESTED** [review]

File: src/engine/DebugOverlays.EntityLabel.cs (hunk 5 above)
Comment (84634E1A607A) [reply src/engine/DebugOverlays.EntityLabel.cs:83]: What? Why?

Comment (Kasterisk) [reply src/engine/DebugOverlays.EntityLabel.cs]: Ah wait, I see that this is still used in line 86. Nevermind this.

**Review by Kasterisk — APPROVED** [review]
LGTM

**Review by hhyyrylainen — APPROVED** [review]
Seems good that the debug panel no longer affects the game microbe objects.

**Pull request merged**